File: schemas/pedido.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class PedidoCreate(BaseModel):
    idClientePed: str
    idRestaurantePed: str
    idEntregadorPed: Optional[str] = None
    valTaxaentregaPed: float
    valTotalPed: float
    dscFormapagammentoPed: str
    dscStatusPed: str


class PedidoResponse(BaseModel):
    idPed: str
    idClientePed: str
    idRestaurantePed: str
    idEntregadorPed: Optional[str] = None
    datPed: datetime
    horPed: datetime
    valTaxaentregaPed: float
    valTotalPed: float
    dscFormapagammentoPed: str
    dscStatusPed: str

File: frontend/src/types/pedido.ts
export interface Pedido {
  idPed: string;
  idClientePed: string;
  idRestaurantePed: string;
  idEntregadorPed?: string;
  datPed: string;
  horPed: string;
  valTaxaentregaPed: number;
  valTotalPed: number;
  dscFormapagammentoPed: string;
  dscStatusPed: string;
}

export interface PedidoCreate {
  idClientePed: string;
  idRestaurantePed: string;
  idEntregadorPed?: string;
  datPed: string;
  horPed: string;
  valTaxaentregaPed: number;
  valTotalPed: number;
  dscFormapagammentoPed: string;
  dscStatusPed: string;
}